Assets/Scripts/VFX/RenderSettings/RenderSettingsManager.cs

```diff
@@ -0,0 +1,193 @@
+	}
+
+	void Awake()
+	{
+		_skyboxTintPropertyID = Shader.PropertyToID( "_Tint" );
+		RenderSettings.skybox.SetFloat( "_Rotation", _skyboxInitRotation );
+
+		_gradientTexture = null;
+		_gradientTexture = new Texture2D( 128, 1, TextureFormat.ARGB32, false );
+
+		UpdateTimeOfDay();
+		UpdateFogGradientTexture();
+	}
+
+	void Update()
+	{
+		RenderSettings.skybox.SetFloat( "_Rotation", Time.realtimeSinceStartup * _skyboxRotSpeed );
+	
+		UpdateRenderSettings();
+	}
+
+	void UpdateTimeOfDay()
+	{
+		float timeOfDay = DayCycleManager.currentTime / ( DayCycleManager.dayCycleLength * ( 1.0f - _nightFraction) );
+		timeOfDay = Mathf.Clamp01( timeOfDay );
+		
+		daylightIntensity = Mathf.Cos( timeOfDay * 2.0f * Mathf.PI ) * -0.5f + 0.5f;
+		
+		TimeLightingSettings.Lerp( _currentLightingSettings.nightSettings,
+		                           _currentLightingSettings.daySettings,
+		                           daylightIntensity,
+		                           ref _currentTimeRenderSettings );
+	}
+
+	void UpdateFogGradientTexture()
+	{
+		Color[] gradientColors = new Color[128];
+		for( int i = 0; i < 128; i++ )
+		{
+			gradientColors[i] = _currentTimeRenderSettings.fogGradient.Evaluate( i/128f );
+		}
+		gradientTexture.SetPixels( gradientColors );
+		gradientTexture.Apply();
+	}
+
+	void UpdateRenderSettings()
+	{
+		UpdateTimeOfDay();
+
+		if( globalFog.fogMaterial != null )
+		{
+			UpdateFogGradientTexture();
+			globalFog.fogMaterial.SetTexture( "_FogGradientTex", gradientTexture );
+		}
+		
+		RenderSettings.skybox.SetColor( _skyboxTintPropertyID, _currentTimeRenderSettings.skyColor );
+		RenderSettings.fogDensity = _currentTimeRenderSettings.fogDensity;
+		
+		_directionalLight.color = _currentTimeRenderSettings.lightColor;
+		_directionalLight.intensity = _currentTimeRenderSettings.lightIntensity;
+	}
+
+	public static void TransitionRenderSettings( LightingSettings newRenderSettings, float shiftTime )
+	{
+		instance.StopAllCoroutines(); // TODO: Make this not a sledgehammer solution
+		instance.StartCoroutine( instance.TransitionRenderSettingsRoutine( newRenderSettings, shiftTime ) );
+	}
+
+	IEnumerator TransitionRenderSettingsRoutine( LightingSettings newRenderSettings, float settingShiftTime )
+	{
+		LightingSettings oldRenderSettings = _currentLightingSettings;
+
+		float settingShiftTimer = 0f;
+		while ( settingShiftTimer < settingShiftTime )
+		{
+			TimeLightingSettings.Lerp( oldRenderSettings.daySettings, 
+			                           newRenderSettings.daySettings,
+			                           settingShiftTimer/settingShiftTime,
+			                           ref _currentLightingSettings.daySettings );
+
+			TimeLightingSettings.Lerp( oldRenderSettings.nightSettings,
+			                           newRenderSettings.nightSettings,
+			                           settingShiftTimer/settingShiftTime,
+			                           ref _currentLightingSettings.nightSettings );
+
+			UpdateRenderSettings();
```

Comment:
Swapped to this. Was running into issues previously where LightingSettings.Lerp was affecting the asset, but that's fixed